accepted/future-releases/small-features-21Q1/feature-specification.md

```diff
@@ -0,0 +1,109 @@
+
+Reintroducing the `>>>` operator was intended for Dart 2.0, but was repeatedly postponed as not important. We do want it for the unsigned shift of integers, and it's been mostly implemented on some of our platforms already. This feature was chosen because it is already half-way implemented.
+
+**Nothing new!** All the chosen changes remove non-orthogonal restrictions on existing features or introduce features analog to other existing features, which means that we do not expect to need a lot of new documentation or educational material. That leaves those who'd write such resources free to deal with the launch of null safety instead.
+
+The changes, in more detail, are detailed below.
+
+## Allow type arguments on annotations
+
+We currently allow metadata to be a call to a constant constructor:
+
+```dart
+@Deprecated("Do not use this thing")
+```
+
+However, the *grammar* does not allow type arguments, meaning that it's not possible to write:
+
+```dart
+@TypeHelper<int>(42, "The meaning")
+```
+
+There is no technical reason for this restriction. It was just simpler, and probably didn't seem necessary at the time metadata was introduced. It does now.
+
+The only change is in the grammar, from
+
+```
+<metadatum> ::= \gnewline{}
+  <identifier> | <qualifiedName> | <constructorDesignation> <arguments>
+```
+
+to
+
+```
+<metadatum> ::= \gnewline{}
+  <identifier> | <qualifiedName> | <constructorDesignation> <argumentPart>
+```
+
+The corresponding constructor invocation must still be valid, now including the type arguments.
+
+The constructed constant, if accessible in any way, will contain the provided type arguments, exactly like if it had been created by a <code>\`const' \<constructorDesignation> \<argumentPart></code> production, and is canonicalized correspondingly.
+
+The largest expected effort for this implementation is the analyzer adding a place to store the type arguments to its public AST. The remaining changes should be using existing functionality.
+
+If type arguments are allowed and omitted, the types are inferred from the types of the arguments to the constructor, as for any other constant invocation. This already happens (checked in VM with `dart:mirrors`), so no change is necessary.
+
+## Allow generic function types as type arguments
+
+The language disallows generic function types as type arguments.
+
+```dart
+List<T Function<T>(T)> idFunctions; // INVALID
+var callback = [foo<T>(T value) => value]; // Inferred as above, then invalid.
+```
+
+We remove that restriction, so a type argument *can* be a generic function type.
+
+This requires no new syntax, and in some cases only the removal of a single check. There might be some platforms where the implementation currently assumes that generic function types cannot occur as the value of type variables (an proof-of-concept attempt hit an assert in the VM). Such assumptions will need to be flushed out with tests and fixed.
```

Comment:
@crelier Do you have any concerns about this?

Reply:
No, I do not have concerns. As mentioned on line 75, tests should point us to the places where we assumed (and hopefully verified with asserts) that we did not have to bother with handling generic types.